src/averageAges.js

```diff
@@ -34,7 +33,13 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  return people.filter(person =>
+    person.sex === 'f'
+    && (withChildren === undefined
+      || (withChildren === true
+        && people.some(child => child.mother === person.name))))
```

Comment:
```suggestion
      || people.some(child => child.mother === person.name))))
```
умова withChildren === true не має ніякого сенсу, оскільки якщо || спрацював і ми дійшли сюди, це вже автоматично значить що withChildren true)